### PR TITLE
PLT-7917: Deactivated user DM improvements

### PR DESCRIPTION
**Pull request by ccbrown**
#### Summary
This implements all of [PLT-7917](https://mattermost.atlassian.net/browse/PLT-7917) except two things:

* It doesn't auto-close DMs. That change is in redux: https://github.com/mattermost/mattermost-redux/pull/338
* I haven't updated the direct message modal yet. I'll do that soon (It can go in a separate PR if needed.).

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-7917

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Has UI changes
- [x] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/mattermost-webapp/blob/master/i18n/en.json)) updates

### CODE REVIEW

**Comment by ccbrown** [comment]
Reassigned since saturn has time off coming up.

File: components/rhs_thread/rhs_thread.jsx [hunk 1]
```diff
@@ -404,6 +406,23 @@ export default class RhsThread extends React.Component {
             );
         }
 
+        const channel = ChannelStore.get(selected.channel_id);
```
Comment (jwilander) [inline components/rhs_thread/rhs_thread.jsx]: We shouldn't add any new uses of the ChannelStore. If you need a channel you should pull it in from a selector [in the container](https://github.com/mattermost/mattermost-webapp/blob/master/components/rhs_thread/index.js) and add it to the component's props

**Comment by ccbrown** [comment]
@jwilander updated!

**Comment by ccbrown** [comment]
@esethna 

1. Not sure how I missed that. Fixed.
2. Yeah I haven't updated that modal yet (see my summary). It's probably best if I put that functionality in with a separate PR so we can go ahead and get this in today.

**Comment by ccbrown** [comment]
@hmhealey just waiting on you!

**Pull request merged**